src/services/cli-http-client.js

```diff
@@ -64,12 +65,16 @@ class CliRequestClient {
       const b64Auth = Buffer.from(`${opts.username}:${opts.password}`).toString('base64');
       headers.Authorization = `Basic ${b64Auth}`;
     }
-
-    const componentInfo = (headers['User-Agent'] || '').replace(' (', '|').replace(')', '').split('|');
-    componentInfo.push(`${os.platform()} ${os.release()} ${os.arch()}`);
+    const componentInfo = [];
+    componentInfo.push(`(${os.platform()} ${os.arch()})`);
+    const userAgentArr = (headers['User-Agent'] || '').split(' ');
+    componentInfo.push(userAgentArr[0]); // extensions
+    componentInfo.push(userAgentArr[3]); // extensions
     componentInfo.push(this.commandName);
     componentInfo.push(this.keytarWord);
-    headers['User-Agent'] = `${pkg.name}/${pkg.version} (${componentInfo.filter(Boolean).join(', ')})`;
+    headers['User-Agent'] = `${this.helperLibrary} ${pkg.name}/${pkg.version} ${componentInfo
+      .filter(Boolean)
+      .join(' ')}`;
 
```

Comment:
If possible, Can we add test for this change?

Reply:
There's no unit test particularly to validate the user-agent header, I could only test it manually with the cli main branch referring to this feature cli-core branch. 
_[DEBUG] User-Agent: @twilio/cli-core/5.27.1 (darwin x64) twilio-node/3.66.1 node/v16.4.2 debugger:logs:list keytar_

Reply:
Have added a couple of test cases [here](https://github.com/twilio/twilio-cli-core/blob/6bbdbce7b530c4c32ab673a6678efb2e8b104a78/test/services/cli-http-client.test.js#L73), please see if it looks good.